packages/block-library/src/template-part/variations.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * WordPress dependencies
+ */
+import { footer, header } from '@wordpress/icons';
+import { store as coreDataStore } from '@wordpress/core-data';
+import { select } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+const createIsActiveBasedOnArea = ( area ) => ( { theme, slug } ) => {
+	if ( ! slug ) {
+		return false;
+	}
+
+	const entity = select( coreDataStore ).getEntityRecord(
+		'postType',
+		'wp_template_part',
+		`${ theme }//${ slug }`
+	);
+
+	return entity?.area === area;
+};
+
+const variations = [
+	{
+		name: 'header',
+		title: __( 'Header' ),
+		icon: header,
+		isActive: createIsActiveBasedOnArea( 'header' ),
```

Comment:
Nitpick: we could add them in a loop like `embed` (ex: https://github.com/WordPress/gutenberg/blob/38f73621335b28af09b3913f52fb38fd4592738e/packages/block-library/src/embed/variations.js#L347)

Reply:
I think we could use the `name` property here as `area`. But I think we should keep it explicit.

Reply:
>I think we could use the name property here as area.

Yes, that's what I meant. Okay go for explicit.